drape_frontend/rule_drawer.cpp

```diff
@@ -457,27 +413,25 @@ void RuleDrawer::operator()(FeatureType & f)
        !ftypes::IsBuildingChecker::Instance()(types)))
     return;
 
+  if (ftypes::IsCoastlineChecker::Instance()(types) && !CheckCoastlines(f))
+    return;
+
   Stylist const s(f, m_zoomLevel, m_deviceLang);
 
   // No drawing rules.
-  if (s.m_symbolRule == nullptr && s.m_captionRule == nullptr && s.m_houseNumberRule == nullptr &&
-      s.m_lineRules.empty() && s.m_areaRule == nullptr && s.m_hatchingRule == nullptr)
-    return;
-
-  if (s.m_isCoastline && !CheckCoastlines(f))
+  if (!s.m_symbolRule && !s.m_captionRule && !s.m_houseNumberRule &&
```

Comment:
What is this case? How often does it happen? Should the higher level code filter it out?

Reply:
It happens when e.g. all feature's drules are not drawable on a given zoom level, normally it means the feature shouldn't be in the visibility index for this zoom. But there could be old mwms, vis changes in styles...
Maybe there are some other cases also.